### PR TITLE
hcm: add downstream termination flag

### PR DESCRIPTION
**Pull request by botengyao**
I think it makes sense to set the `DownstreamConnectionTermination` flag when getting `remote` and `local` close.

remote close actually is set here but no harm to set for both: https://github.com/envoyproxy/envoy/blob/a41d9ad6f8e0c8f8b92c8f7daf8b6c9d810b1bfa/source/common/http/conn_manager_impl.cc#L950-L954

Commit Message:
Additional Description:
Risk Level: low.
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/common/http/conn_manager_impl.cc
Comment (alyssawilk) [inline source/common/http/conn_manager_impl.cc:601]: should we then remove the now redundant setting of  remote_close in ConnectionManagerImpl::ActiveStream::completeRequest() { ?  Not sure if it's safe to move the response code details setting inline but it seems plausible we could clean both up.

Also can you add a comment at the definition of DownstreamConnectionTermination to disambiguate it's (now) the termination of the downstream connection not a connection termination initiated by downstream?

**Review by alyssawilk — COMMENTED** [review]
LGTM modulo the 2 nits below.  Mind adding a !google reviewer as well?  doesn't need to be a senior maintainer I just want a non-google cross-check.
/wait

File: source/common/http/conn_manager_impl.cc
Comment (botengyao) [reply source/common/http/conn_manager_impl.cc:612]: Thanks for the quick review! 

Good point, I think it is safe and removed it. The comment right now is `// If the stream was reset due to a downstream connection termination.`, which looks reasonable to me.

**Comment by botengyao** [comment]
/assign @soulxu 

for cross-check, thanks!

**Review by soulxu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
